test/fixtures/poro.rb

```diff
@@ -266,3 +266,18 @@ def json_key
     raise StandardError, 'Intentional error for rescue_from test'
   end
 end
+
+CommentNestedSerializer = Class.new(ActiveModel::Serializer) do
+  attributes :body, :id
+end
+
+PostNestedSerializer = Class.new(ActiveModel::Serializer) do
+  attributes :title, :body, :id
+  has_many :comments, serializer: CommentNestedSerializer
+end
+
+AuthorNestedSerializer = Class.new(ActiveModel::Serializer) do
+  attributes :id, :name
+  has_many :posts, serializer: PostNestedSerializer
+end
```

Comment:
I have a suggestion.
In my projects, I am doing like this way:

```ruby
class Author < ActiveModel::Serializer
  attributes :id, :name

  class ForShow < self
    has_many :posts
  end
end

class Post < ActiveModel::Serializer
  attributes :id, :title, :body

  class ForShow < self
    has_many :comments
  end
end
```

and so on.

I define default serializers not to have any  relations.
And for specific cases, like showing details, define serializers as nested classes.

When I use them in a controller, I write like:
```ruby
def show
  ...
  render json: @post, serializer: PostSerializer::ForShow
end
```

Like this way, I can easily avoid unexpected circular relations.
Also scoping with default serializer classes make it clear which serializer is for which model.